### PR TITLE
Only show zwave_js firmware action if no other updates in progress

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
We only want one upgrade to be going on at a time. This means that we should only show the action for a node if the firmware is upgradeable for that node, there are no nodes currently doing updates, or if there are, it's the current node.

Dependent on https://github.com/home-assistant/core/pull/73905

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
Core PR is approved.

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts [hunk 1]
```diff
@@ -88,7 +90,11 @@ export const getZwaveDeviceActions = async (
   const firmwareUpdateCapabilities =
     await fetchZwaveNodeFirmwareUpdateCapabilities(hass, device.id);
 
-  if (firmwareUpdateCapabilities.firmware_upgradable) {
+  if (
+    firmwareUpdateCapabilities.firmware_upgradable &&
+    (!(await fetchZwaveIsAnyFirmwareUpdateInProgress(hass, entryId)) ||
+      (await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id)))
+  ) {
```
Comment (zsarnett) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts:97]: Can you do this above? So its easier to read?
```suggestion
  const anyUpdatesInProgress = await fetchZwaveIsAnyFirmwareUpdateInProgress(hass, entryId);
  const deviceUpdatesInProgress = await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id);
  
  if (
    firmwareUpdateCapabilities.firmware_upgradable &&
    (!entryUpdatesInProgress || deviceUpdatesInProgress)
  ) {
```

Something like the above

**Review by zsarnett — DISMISSED** [review]
1 comment

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts (hunk 1 above)
Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts:97]: yup, I was trying to save a potential call

Comment (zsarnett) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts]: You could still do that. in the var. In the second one only do the call if any updates is false.

```ts
const deviceUpdatesInProgress = anyDeviceUpdatesInProgress ? true : await fetchZwaveNodeIsFirmwareUpdateInProgress(hass, device.id);
```

Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts]: honestly it's probably more performant to fire all three calls and just await all of them right? (e.g. how I did it)

**Review by zsarnett — APPROVED** [review]

**Comment by raman325** [comment]
core PR merged

**Pull request merged**